back-end/helpers/intent_analyzer.spec.js

```diff
@@ -0,0 +1,32 @@
+import testIntentGenerator from "./test-helpers/testIntentGenerator.js";
+import intentAnalyzer from "./intent_analyzer.js"
+
+let generator;
+
+describe("intentAnalyzer", () => {
```

Comment:
I think there is a major issue with all of these tests.

Ok so first off, the reason why the first test is failing is because `formatIntents` in `intentAnalyzer` is marked as `async` so it returns a Promise. Until this Promise is resolved, it is not actually the same as `{ }`.

I managed to get the test to "pass" by creating a new async method in the generator to return a `{ }`, and then changing the test to compare against that method's return, since they would both be Promises of a `{ }`.

But then I realized that since both `formatIntents` and the generator methods are returning unresolved Promises, they are not actually comparing equality in the return values. I have just tried to completely mess up the inputs and outputs in the generator, which should fail the tests, but instead they pass because they are only checking the Promises.

I think you may be able to resolve promises into their values using `.resolve()`, but for some reason I can't get it work. So maybe look more into that or better would be to stop using async since that is overcomplicating things. And in general, for test-writing you should try purposely failing your tests to make sure they are actually checking properly.

Reply:
There is more going wrong here I think, but the core thing wrong is that we are not awaiting our results. .resolve is not needed. Great intuition on the tests failing!